lib/tests/test_merged_tree.rs

```diff
@@ -278,3 +279,120 @@ fn test_resolve_with_conflict() {
         Conflict::new(vec![expected_base1], vec![expected_side1, expected_side2])
     )
 }
+
+#[test]
+fn test_conflict_iterator() {
```

Comment:
Re commit description:

>  We assume
that the trees have previously been resolved as much as possible, so
we don't attempt to resolve conflicts again.


I'm not sure how important this is, but you could have an example where there is a 3-sided conflict and a 2-sided conflict in the same tree. This could happen in two ways:

- "normal" way where there are 3 trees in the conflcit.
- from a "legacy" tree, where the missing sides would be replaced by `None`.

I'm not 100% sure what will happen. Sorry if you have a test like that already and I missed it.